main.py

```diff
@@ -129,30 +129,27 @@ def show_settings(self):
             self.dialog.findChild(QtCore.QObject, RECORD_SETTINGS).setChecked(True)
         if cfg[CONFIG_MUTE]:
             self.dialog.findChild(QtCore.QObject, MUTE_SETTINGS).setChecked(True)
-        index = self.dialog.findChild(QtCore.QObject, QUALITY_SETTINGS).findText(cfg[CONFIG_QUALITY])
-        self.dialog.findChild(QtCore.QObject, QUALITY_SETTINGS).setCurrentIndex(index)
+        self.dialog.findChild(QtCore.QObject, QUALITY_SETTINGS).setText(", ".join(cfg[CONFIG_QUALITY]))
         self.dialog.findChild(QtCore.QObject, BUFFER_SIZE).setValue(cfg[CONFIG_BUFFER_SIZE])
         self.dialog.show()
 
     def generate_conf(self):
-        """Reads values from settings and generates new config"""
-        is_buffer = False
-        is_mute = False
-        quality = str(self.dialog.findChild(QtCore.QObject, QUALITY_SETTINGS).currentText())
-        buffer_size = self.dialog.findChild(QtCore.QObject, BUFFER_SIZE) \
-            .value()
-        if self.dialog.findChild(QtCore.QObject, RECORD_SETTINGS).isChecked():
-            is_buffer = True
-        if self.dialog.findChild(QtCore.QObject, MUTE_SETTINGS).isChecked():
-            is_mute = True
+        """Generate a config from the provided settings."""
+        # Quality options should be provided as a comma-separated string.
+        quality_text = self.dialog.findChild(QtCore.QObject, QUALITY_SETTINGS).text()
+        quality_options = [quality.strip() for quality in quality_text.split(",")]
 
         # Set new cfg values
-        cfg[CONFIG_BUFFER_STREAM] = is_buffer
-        cfg[CONFIG_MUTE] = is_mute
-        cfg[CONFIG_QUALITY] = quality
-        cfg[CONFIG_BUFFER_SIZE] = buffer_size
+        cfg[CONFIG_QUALITY] = quality_options
+        cfg[CONFIG_BUFFER_STREAM] = self.dialog.findChild(QtCore.QObject, RECORD_SETTINGS).isChecked()
+        cfg[CONFIG_MUTE] = self.dialog.findChild(QtCore.QObject, MUTE_SETTINGS).isChecked()
+        cfg[CONFIG_BUFFER_SIZE] = self.dialog.findChild(QtCore.QObject, BUFFER_SIZE).value()
 
-        QtWidgets.QMessageBox.critical(self, "Caution", "Some of your changes may require a restart in order to take effect.")
+        QtWidgets.QMessageBox.critical(
```

Comment:
Maybe it's better UX to tell the user to restart the application regardless? Now they might think that a config has taken effect when it actually requires a restart, which could lead to frustration since they don't know which configs require restart.